tests/Integration/Behaviour/Features/Context/Domain/Product/Combination/DefaultCombinationFeatureContext.php

```diff
@@ -0,0 +1,184 @@
+     */
+    public function assertDefaultCombinationForShop(
+        string $productReference,
+        string $shopReference,
+        string $combinationReference
+    ): void {
+        $this->assertDefaultCombination(
+            $productReference,
+            $combinationReference,
+            $this->getSharedStorage()->get($shopReference)
+        );
+    }
+
+    /**
+     * @Given product :productReference should not have a default combination
+     * @Given product :productReference does not have a default combination
+     *
+     * @param string $productReference
+     */
+    public function assertProductHasNoCachedDefaultCombination(string $productReference): void
+    {
+        $this->assertCachedDefaultCombinationId($productReference, 0, $this->getDefaultShopId());
+    }
+
+    /**
+     * @Given product :productReference should not have a default combination for shop ":shopReference"
+     *
+     * @param string $productReference
+     */
+    public function assertProductHasNoCachedDefaultCombinationForShop(string $productReference, string $shopReference): void
+    {
+        $this->assertCachedDefaultCombinationId(
+            $productReference,
+            0,
+            $this->getSharedStorage()->get($shopReference)
+        );
+    }
+
+    /**
+     * @param string $productReference
+     * @param int $combinationId
+     */
+    private function assertCachedDefaultCombinationId(string $productReference, int $combinationId, int $shopId): void
+    {
+        $product = new Product(
+            $this->getSharedStorage()->get($productReference),
+            false,
+            null,
+            $shopId
+        );
+
+        Assert::assertEquals(
+            $combinationId,
+            (int) $product->cache_default_attribute,
+            'Unexpected cached product default combination'
+        );
+    }
+
+    /**
+     * @param string $productReference
+     * @param string $combinationReference
+     * @param int $shopId
+     */
+    private function assertDefaultCombination(
+        string $productReference,
+        string $combinationReference,
+        int $shopId
+    ) {
+        $combinationId = $this->getSharedStorage()->get($combinationReference);
+
+        $this->assertCachedDefaultCombinationId(
+            $productReference,
+            $combinationId,
+            $shopId
+        );
+
+        Assert::assertTrue(
+            $this->getCombinationForEditing($combinationReference, $shopId)->isDefault(),
+            sprintf('Unexpected default combination in CombinationForEditing for "%s"', $combinationReference)
+        );
+    }
+
+    /**
+     * @param string $combinationReference
+     * @param ShopConstraint $shopConstraint
+     */
+    private function setDefaultCombination(string $combinationReference, ShopConstraint $shopConstraint): void
+    {
+        try {
+            $this->getCommandBus()->handle(new SetDefaultCombinationCommand(
```

Comment:
Now that we have the unified command I'm thinking maybe this command will have to be removed in favor of the unified one What do you think?